test_knight.py

```diff
@@ -0,0 +1,161 @@
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY]
+    ]
+    game.board[3][4] = k1
+    game.board[1][3] = k2
+    assert sorted(k1.get_valid_peaceful_moves(game)) == [(1, 5), (2, 2), (2, 6), (4, 2), (4, 6), (5, 3), (5, 5)]
+
+
+def test_get_valid_piece_takes_empty():
+    game = game_state()
+    k1 = Knight('n', 3, 4, Player.PLAYER_1)
+    game.board = board = [
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY]
+    ]
+    game.board[3][4] = k1
+    assert sorted(k1.get_valid_piece_takes(game)) == []
+
+
+def test_get_valid_piece_takes():
+    game = game_state()
+    k1 = Knight('n', 3, 4, Player.PLAYER_1)
+    k2 = Knight('n', 1, 3, Player.PLAYER_2)
+    game.board = board = [
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
+        [Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY, Player.EMPTY,
+         Player.EMPTY],
```

Comment:
You are creating this board for each function, try making the init of the board a function as well so the code would be simpler to read.

Reply:
You are correct. i created function to init the board to empty board